community/kernel-api/src/main/java/org/neo4j/internal/kernel/api/Session.java

```diff
@@ -19,12 +19,30 @@
  */
 package org.neo4j.internal.kernel.api;
 
+import org.neo4j.internal.kernel.api.exceptions.KernelException;
+
 /**
  * A Session is used to gain
  */
 public interface Session extends AutoCloseable
 {
-    Transaction beginTransaction();
+    /**
+     * Begin new explicit transaction.
+     *
+     * @return The new transaction
+     */
+    default Transaction beginTransaction() throws KernelException
+    {
+        return beginTransaction( Transaction.Type.explicit );
```

Comment:
I don't think default implementations should be used in this abstract class kind of style. Implementors are capable of doing this themselves.